src/main/java/de/tum/in/www1/artemis/service/connectors/localci/SharedQueueManagementService.java

```diff
@@ -97,6 +97,14 @@ public List<LocalCIBuildJobQueueItem> getProcessingJobsForCourse(long courseId)
         return processingJobs.values().stream().filter(job -> job.courseId() == courseId).toList();
     }
 
+    public List<LocalCIBuildJobQueueItem> getQueuedJobsForParticipation(long participationId) {
+        return queue.stream().filter(job -> job.participationId() == participationId).toList();
+    }
+
+    public List<LocalCIBuildJobQueueItem> getProcessingJobsForParticipation(long participationId) {
+        return processingJobs.values().stream().filter(job -> job.participationId() == participationId).toList();
+    }
```

Comment:
Instead of collecting a list and then checking if the list is empty, you could also directly return booleans here and check if there exists a value in the stream.